kibble/render/render.go

```diff
@@ -97,29 +95,56 @@ func Render(sourcePath string, buildPath string, cfg *models.Config) int {
 	renderer.Initialise()
 
 	initSW.Completed()
-
 	errCount := 0
-
 	renderSW := utils.NewStopwatchLevel("render", logging.NOTICE)
-	for lang, localeObj := range cfg.Languages {
 
-		locale := localeObj.Code
+	// Use data from APIs if site_translations_api toggle is enabled.
+	// Otherwise, use data from kibble.json.
+	var defaultLanguage string
+	var languageConfigs map[string]models.LanguageConfig
+	var translationFilePath string
+
+	var useApi = site.Toggles["site_translations_api"]
+	log.Infof("UseTranslationsApi: %t", useApi)
+
+	if useApi {
+		defaultLanguage = site.DefaultLanguage
+		languageConfigs = site.LanguagesToLanguageConfigs()
+		translationFilePath = buildPath
+		//Setup language files for writing translations obtained by API
+		err = ensureLanguageFiles(site, buildPath)
+		if err != nil {
+			return 1
+		}
+
+	} else {
+		defaultLanguage = cfg.DefaultLanguage
+		languageConfigs = cfg.Languages
+		translationFilePath = sourcePath
+	}
+
+	for languageKey, language := range languageConfigs {
+
+		code := language.Code
 
 		ctx := models.RenderContext{
 			RoutePrefix: "",
 			Site:        site,
-			Language:    createLanguage(cfg, lang, locale),
+			Language:    createLanguage(defaultLanguage, languageKey, code),
 		}
 
-		if lang != cfg.DefaultLanguage {
-			ctx.RoutePrefix = fmt.Sprintf("/%s", lang)
-			i18n.LoadTranslationFile(filepath.Join(sourcePath, ctx.Language.DefinitionFilePath))
+		if languageKey != defaultLanguage {
+			ctx.RoutePrefix = fmt.Sprintf("/%s", languageKey)
+			err := i18n.LoadTranslationFile(filepath.Join(translationFilePath, ctx.Language.DefinitionFilePath))
+			if err != nil {
+				log.Errorf("Translation file load failed: %s", err)
+			}
 		} else {
-			i18n.MustLoadTranslationFile(filepath.Join(sourcePath, ctx.Language.DefinitionFilePath))
+			i18n.MustLoadTranslationFile(filepath.Join(translationFilePath, ctx.Language.DefinitionFilePath))
 		}
 
-		renderLangSW := utils.NewStopwatchf("  render language: %s", lang)
-		T, err := i18n.Tfunc(locale, cfg.DefaultLanguage)
+		renderLangSW := utils.NewStopwatchf("  render language: %s", languageKey)
```

Comment:
Is this worth having a stop watch if there's no completed fn?

Reply:
It completes on line 167 right?